### DISCUSSION TITLE
Show distance / time at all locations on the route

### DISCUSSION THREAD

**Discussion by MotoUKRider** [devemux86/cruiser#694] [Q&A] [answered]
([from this discussion](https://github.com/devemux86/cruiser/discussions/692))

When touching a route line, some way to easily see where/when you are in distance+time - more so if there are no ViaPoints to work with.

This would be especially useful on unknown roads where 50mi on a mountain pass would be a longer (and more challenging) ride than a 50mi motorway sprint.

**Comment by devemux86** [comment]
@MotoUKRider Thanks for the suggestion.

Something like this seems possible, I will check it.

**Comment by devemux86** [comment]
This is already available with great accuracy with the route diagram.

Tap (or drag) on the diagram to see the distance and elevation at each location on the route:

<img src="https://github.com/user-attachments/assets/07fc2eb5-8398-436d-8e29-fd2f33a23375" width="50%"/>

**Reply by MotoUKRider** [reply]
I was hoping you wouldn't notice that method! It's clutter I can do without on a small screen.

Also, this is a Config option rather than an easy thing to toggle on/off, right?

**Reply by devemux86** [reply]
@MotoUKRider I don't understand what do you mean?

Long press on route line on a small screen is much less accurate than drag on the diagram to move precisely the crosshair.
It has the same great accuracy of the crosshair when we add or move waypoints.

**Reply by devemux86** [reply]
@MotoUKRider You can also rotate your phone to landscape.
Then the route diagram is much longer with more room for gestures.

**Reply by MotoUKRider** [reply]
Compromise: Allow the turn points to offer the position in their bubble popup, ie: add a 2nd line that shows the distance/time overall.

That would allow a workflow like: *"So how far into the journey do we turn left at the A123?"*  Accuracy would be assured as Cruiser knows what we've tapped.

**Reply by devemux86** [reply]
@MotoUKRider This will complicate everything and will still only be at fixed points.

The request is to see the data everywhere on the route, not only at fixed points.
Otherwise we could display distance / time on shaping points too.

**Reply by devemux86** [reply]
Only alternative to route diagram seems to be the screen when we long press anywhere on the route.
Or the toast message when we tap anywhere on the route line.

But as the space is limited, then we will display the distance / time of the point and not of whole route.

**Reply by MotoUKRider** [reply]
I'm utterly confuzzled as to why a single press is "complicating everything" when your alternative is a 17-point tapping exercise?  If it's not practical to offer it *anywhere* along the route due to accuracy (and/or dialog space) then my compromise idea seemed like the next best thing.

Surely there are others who feel that tapping on a turn instruction icon and it producing the route distance+time is a handy thing?

**Reply by devemux86** [reply]
@MotoUKRider Please allow the developers to know better the complexity of the apps.

As you know better how difficult it is to maintain a map theme compared to users. :slightly_smiling_face:

**Comment by devemux86** [comment]
As a first step we can also display the distance / time at the shaping points:

<img src="https://github.com/user-attachments/assets/b9e3a558-2acf-4d99-a409-a081257abfe5" width="50%"/><img src="https://github.com/user-attachments/assets/d3c24b1f-eed8-4410-bf00-60f49a1939e0" width="50%"/>

**Comment by devemux86** [comment]
When long press on the route, the route menu will display the distance/time from the location to the start and destination:

<img src="https://github.com/user-attachments/assets/05e593e2-7954-4d6d-952c-ccd8b625a9b5" width="50%"/>

**Reply by devemux86** [reply]
Similarly on the desktop (with right click).

You can see how much better this solution is since we can get the distance / time at all locations.
(compared to turn instructions that are sparse on the route)

![Cruiser](https://github.com/user-attachments/assets/18c317bb-f81a-4b3c-aaa1-bc607ade62df)

**Reply by bios4** [reply]
> display the distance/time from the location to the start and destination

What represents what?

<img src="https://github.com/user-attachments/assets/32e220b9-7f45-44a4-8ac3-ffd3bc5717f1" width="33%">

It would be great if there could be some indicators, e.g. a "home" icon and a "final flag" or something...
Is it possible?

**Reply by devemux86** [reply]
Distance / time to start - Distance / time to end.
You mean like the updated screenshot [above](https://github.com/devemux86/cruiser/discussions/694#discussioncomment-11628012)?

**Reply by bios4** [reply]
> You mean like the updated screenshot above?

Exactly - except that “arrow-up” and “arrow-down” are not necessarily unambiguous as to what is meant.

To be honest, I would e.g. interpret the "arrow-down" as "time to start", because it lies "behind me" and the arrow down points in the opposite direction to where I'm going.

Here, for example, I would find (Material Icons code) `home` or `house` (start) and `flag` or `sports_score` (destination) more meaningful, if possible...

**Reply by devemux86** [reply]
@bios4 This is text, we need unicode characters (not icons).
We use the same in waypoints list and info bubbles.

**Reply by Ciutnik** [reply]
I remember we once had a discussion on the occasion of the new road book, about symbols for the start/end and next/previous points. The arrows are fine, they are aesthetic, they take up little space and once you understand their meaning you will never have any doubts about their meaning 🙂

**Reply by devemux86** [reply]
It also depends where we display the data. If the field allows icons or only text.

**Reply by MotoUKRider** [reply]
The above examples look good. Having it on Desktop is an added bonus.

I was thinking that the title of the dialog may now be confusing (not showing the full route stats), but it now actually follows the same principle of touching any other area where you see the Lat/Lon and a series of panels, eg: "What do you want to do with *this* location?"

**Reply by devemux86** [reply]
> it now actually follows the same principle of touching any other area where you see the Lat/Lon and a series of panels, eg: "What do you want to do with this location?"

Exactly that. :slightly_smiling_face:

**Comment by devemux86** [comment]
> When long press on the route, the route menu will display the distance/time from the location to the start and destination

> we can also display the distance / time at the shaping points

Available in the latest:
- https://github.com/devemux86/cruiser/discussions/690

**Reply by MotoUKRider** [reply]
Nice work, thanks!

**Comment by devemux86** [comment, marked as answer]
- Show distance / time at all locations on the route is available in https://github.com/devemux86/cruiser/discussions/701
